common/achievement/aggregator.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { aggregators } from './aggregator';
+
+describe('test sum aggregator', () => {
+    const tests: {
+        name: string;
+        elements: number[];
+        expected: number;
+    }[] = [
+        {
+            name: 'should sum all elements in array',
+            elements: [1, 2, 3, 4, 5],
+            expected: 15,
+        },
+        {
+            name: 'should return 0 for empty array',
+            elements: [],
+            expected: 0,
+        },
+    ];
+
+    it.each(tests)('$name', ({ elements, expected }) => {
+        expect(aggregators['sum'].function(elements)).toEqual(expected);
+    });
+});
+
+describe('test count aggregator', () => {
+    const tests: {
+        name: string;
+        elements: number[];
+        expected: number;
+    }[] = [
+        {
+            name: 'should count all elements in array',
+            elements: [1, 2, 3, 4, 5],
+            expected: 5,
+        },
+        {
+            name: 'should return 0 for empty array',
+            elements: [],
+            expected: 0,
+        },
+        {
+            name: 'should skip elements that are zero',
+            elements: [1, 0, 3, 0, 5],
+            expected: 3,
+        },
+    ];
+
+    it.each(tests)('$name', ({ elements, expected }) => {
+        expect(aggregators['count'].function(elements)).toEqual(expected);
+    });
+});
+
+describe('test presenceOfEvents aggregator', () => {
+    const tests: {
+        name: string;
+        elements: number[];
+        expected: number;
+    }[] = [
+        {
+            name: 'should return 1 if at least one element is present',
+            elements: [1, 2, 3, 4, 5],
+            expected: 1,
+        },
+        {
+            name: 'should return 0 for empty array',
+            elements: [],
+            expected: 0,
+        },
+        {
+            name: 'should return 0 if all elements are zero',
+            elements: [0, 0, 0, 0, 0],
+            expected: 0,
+        },
+    ];
+
+    it.each(tests)('$name', ({ elements, expected }) => {
+        expect(aggregators['presenceOfEvents'].function(elements)).toEqual(expected);
+    });
+});
+
+describe('test lastStreakLength aggregator', () => {
+    const tests: {
+        name: string;
+        elements: number[];
+        expected: number;
+    }[] = [
+        {
+            name: 'should get a streak of five',
+            elements: [1, 2, 3, 4, 5],
+            expected: 5,
+        },
+        {
+            name: 'should not have a streak if no elements are present',
+            elements: [],
+            expected: 0,
+        },
+        {
+            name: 'should reset streak if zero in chain',
+            elements: [1, 1, 0, 1, 1],
```

Comment:
How is the chain sorted? Maybe change to [1, 0, 1, 1] to also check the order?

Reply:
The next test is ensuring that the evaluation oder is correct. Nevertheless, I'll update this test to do the same.